sdk/src/v2_api/reader.rs

```diff
@@ -0,0 +1,267 @@
+    /// # Example
+    /// ```no_run
+    /// use c2pa::Reader;
+    /// let bytes = include_bytes!("../../tests/fixtures/CA.jpg");
+    /// let reader = Reader::from_bytes("image/jpeg", bytes).unwrap();
+    /// ```
+    pub fn from_bytes(format: &str, bytes: &[u8]) -> Result<Reader> {
+        Self::from_stream(format, Cursor::new(bytes))
+    }
+
+    #[cfg(feature = "file_io")]
+    /// Create a manifest store Reader from a file
+    /// # Arguments
+    /// * `path` - The path to the file
+    /// # Returns
+    /// A reader for the manifest store
+    /// # Errors
+    /// If the file is not a valid manifest store
+    /// validation status should be checked for non severe errors
+    /// # Example
+    /// ```no_run
+    /// use c2pa::Reader;
+    /// let reader = Reader::from_file("path/to/file.jpg").unwrap();
+    /// ```
+    /// # Note
+    /// If the file does not have a manifest store, the function will check for a sidecar manifest
+    /// with the same name and a .c2pa extension
+    /// If the sidecar manifest exists, it will be used instead
+    pub fn from_file<P: AsRef<std::path::Path>>(path: P) -> Result<Reader> {
+        let path = path.as_ref();
+        let format = crate::format_from_path(path).ok_or(crate::Error::UnsupportedType)?;
+        let mut file = File::open(path)?;
+        let result = Self::from_stream(&format, &mut file);
+        if let Err(Error::JumbfNotFound) = result {
+            // if not embedded or cloud, check for sidecar first and load if it exists
+            let potential_sidecar_path = path.with_extension("c2pa");
+            if potential_sidecar_path.exists() {
+                let manifest_data = read(potential_sidecar_path)?;
+                return Self::from_manifest_data_and_stream(&manifest_data, &format, &mut file);
+            }
+        }
+        result
+    }
+
+    /// Create a manifest store Reader from a JSON string
+    /// # Arguments
+    /// * `json` - The JSON string
+    /// # Returns
+    /// A reader for the manifest store
+    /// # Note
+    /// This should only be used for testing
+    /// Any referenced resources will not be available
+    pub fn from_json(json: &str) -> Result<Reader> {
+        let manifest_store = serde_json::from_str(json)?;
+        Ok(Reader { manifest_store })
+    }
+
+    /// Create a manifest store Reader from existing c2pa_data and a stream
+    /// You can use this to validate a remote manifest or a sidecar manifest
+    /// # Arguments
+    /// * `c2pa_data` - The c2pa data (a manifest store in JUMBF format)
+    /// * `format` - The format of the stream
+    /// * `stream` - The stream to verify the store against
+    /// # Returns
+    /// A reader for the manifest store
+    /// # Errors
+    /// If the c2pa_data is not valid, or severe errors occur in validation
+    /// validation status should be checked for non severe errors
+    #[async_generic(async_signature(
+        c2pa_data: &[u8],
+        format: &str,
```

Comment:
I noticed that `from_manifest_data_and_stream` is `(c2pa_data, format)` but `from_bytes` is `(format, bytes)` - IMO it would be good to always have `format` first or always have it after the input data argument to stay consistent.

Reply:
That's because the format in this case applies to the stream, not the c2pa_data. So format still comes first. I suppose I could make c2pa_data the last parameter if that would help.